test/should_pass/user_types.erl

```diff
@@ -0,0 +1,15 @@
+-module(user_types).
+
+-export([f/1]).
+
+%% these exported types are used in remote_types module
+-export_type([my_tuple/0, my_list/0, my_union/0]).
+
```

Comment:
So this file is a test of it's own as well as a dependency for another test. That's interesting... I think it's fine. It's good that it is noted in the comment.

Reply:
This is correct.
These should pass/should fail files are very handy to add just random examples, but a bit harder to add well structured test cases covering many variations. But at this stage of the project it is good enough.